pydocx/export/html.py

```diff
@@ -142,268 +152,711 @@ def style(self):
-        )
+    def export_run_property_vanish(self, run, results):
+        attrs = {
+            'class': 'pydocx-hidden',
+        }
+        tag = HtmlTag('span', **attrs)
+        return tag.apply(results, allow_empty=False)
 
-    def page_break(self):
-        return '<hr />'
-
-    def _convert_measurement(self, value):
-        '''
-        >>> exporter = PyDocXHTMLExporter('foo')
-        >>> exporter._convert_measurement(30)
-        0.125
-        '''
-        return value / TWIPS_PER_POINT / POINTS_PER_EM
-
-    def indent(
-        self,
-        text,
-        alignment=None,
-        firstLine=None,
-        left=None,
-        right=None,
-    ):
+    def export_run_property_hidden(self, run, results):
+        attrs = {
+            'class': 'pydocx-hidden',
+        }
+        tag = HtmlTag('span', **attrs)
+        return tag.apply(results, allow_empty=False)
+
+    def export_run_property_vertical_align(self, run, results):
+        if run.effective_properties.is_superscript():
+            return self.export_run_property_vertical_align_superscript(
+                run,
+                results,
+            )
+        elif run.effective_properties.is_subscript():
+            return self.export_run_property_vertical_align_subscript(
+                run,
+                results,
+            )
+        return results
+
+    def export_run_property_vertical_align_superscript(self, run, results):
+        tag = HtmlTag('sup')
+        return tag.apply(results, allow_empty=False)
+
+    def export_run_property_vertical_align_subscript(self, run, results):
+        tag = HtmlTag('sub')
+        return tag.apply(results, allow_empty=False)
+
+    def export_text(self, text):
+        results = super(PyDocXHTMLExporter, self).export_text(text)
+        for result in results:
+            if result:
+                yield result
+
+    def export_deleted_text(self, deleted_text):
+        # TODO deleted_text should be ignored if it is NOT contained within a
+        # deleted run
+        results = self.export_text(deleted_text)
+        attrs = {
+            'class': 'pydocx-delete',
+        }
+        tag = HtmlTag('span', **attrs)
+        return tag.apply(results, allow_empty=False)
+
+    def get_hyperlink_tag(self, hyperlink):
+        target_uri = hyperlink.get_target_uri()
+        if target_uri:
+            href = self.escape(target_uri)
+            return HtmlTag('a', href=href)
+
+    def export_hyperlink(self, hyperlink):
+        results = super(PyDocXHTMLExporter, self).export_hyperlink(hyperlink)
+        tag = self.get_hyperlink_tag(hyperlink)
+        if tag:
+            results = tag.apply(results, allow_empty=False)
+
+        # Prevent underline style from applying by temporarily monkey-patching
+        # the export unliner function. There's got to be a better way.
+        old = self.export_run_property_underline
+        self.export_run_property_underline = lambda run, results: results
```

Comment:
One alternative is the underline can detect if there's a hyperlink parent, and bail if there is.